### PR TITLE
Use nextcloud-password-confirmation

### LINKED ISSUE #11224: Password input time for admin reaproval to short
When changing settings for nextcloud via web interface the timeframe is to short to type in ma password.
This happens if an session exists a couple of time and reaproval of credentials is forced.

In case you have an very long (and hopefully a secure) password the timeframe to type in exeeds an you face an error message.
This feature has been introduced in NC14.

I've found no option to extend the timeframe for the password input in the web interface.

Please make it configureable.

### PR DESCRIPTION
**Pull request by skjnldsv**
Fix #11224 
~~Needs https://github.com/ChristophWurst/nextcloud-password-confirmation/pull/1~~

### CODE REVIEW

File: core/js/js.js [hunk 1]
```diff
@@ -1775,6 +1775,10 @@ OC.PasswordConfirmation = {
 				function (result, password) {
 					if (result && password !== '') {
 						self._confirmPassword(password, config);
+					} else {
```
Comment (ChristophWurst) [inline core/js/js.js]: `} else if {`

File: core/js/js.js [hunk 2]
```diff
@@ -1775,6 +1775,10 @@ OC.PasswordConfirmation = {
...
+						if (rejectCallback && typeof rejectCallback === 'function') {
+							rejectCallback()
```
Comment (ChristophWurst) [inline core/js/js.js]: as discussed, please use `_.isFunction` here

**Review by ChristophWurst — DISMISSED** [review]
Nice!

**Review by ChristophWurst — APPROVED** [review]
The nicest!

Haven't tested, but code looks good :+1:

**Review by blizzz — APPROVED** [review]
shiny :+1: tested, working.

**Review by JanDragon — APPROVED** [review]
Code looks good.

**Comment by ChristophWurst** [comment]
@MorrisJobke one does not simply merge fixup commits

**Comment by ChristophWurst** [comment]
@rullzer guess we need that CI check you once mentioned ;)

**Comment by rullzer** [comment]
See ;) told you!

**Comment by MorrisJobke** [comment]
> @MorrisJobke one does not simply merge fixup commits

Then don't set "4 - To release" if the history is not ready ;)

**Comment by ChristophWurst** [comment]
> [blizzz](/blizzz) added [4. to release](/nextcloud/server/labels/4.%20to%20release)

:see_no_evil:

**Comment by blizzz** [comment]
Didn't review commit messages, only code :tongue:

**Pull request merged**